pkg/gardenlet/controller/shootstate/extensions/reconciler.go

```diff
@@ -67,17 +65,17 @@ func (r *Reconciler) Reconcile(ctx context.Context, request reconcile.Request) (
 		return reconcile.Result{}, err
 	}
 
-	if shouldSkipExtensionObjectSync(extensionObj) {
-		return reconcile.Result{}, nil
-	}
-
 	var (
 		name         = extensionObj.GetName()
 		purpose      = extensionObj.GetExtensionSpec().GetExtensionPurpose()
 		newState     = extensionObj.GetExtensionStatus().GetState()
 		newResources = extensionObj.GetExtensionStatus().GetResources()
 	)
 
+	if obj.GetDeletionTimestamp() != nil {
```

Comment:
previously when deletion time was set it used to return `reconcile.Result{}, nil`. But now it rather set the resource to nil which will cause it to remove from shootstate by `updateShootStateExtensionStateAndResources` function, is this expected?

Reply:
I changed this intentionally, at least. I don't know why this was ignored before, to me this did not make sense to leave resources orphaned in the ShootState.

Reply:
Thanks!